### ISSUE TITLE
[BUG] `rm.all` doesn't delete anything on Windows

### ISSUE THREAD

**Issue by peterhirn** [npm/cacache#165]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

- version 17.0.4
- node 19.6
- this is a windows only bug, works as expected in `node:19.6.0-alpine`

```js
import cacache from "cacache"

const cachePath = "./cache"
const key = "test-key"

await cacache.put(cachePath, key, "10293801983029384")

const cached = await cacache.get(cachePath, key)
console.log(`Data: ${cached.data}`)

await cacache.rm.all(cachePath)

const cached2 = await cacache.get(cachePath, key)
console.log(`Data after rm.all: ${cached2.data}`)
```

Output
```
Data: 10293801983029384
Data after rm.all: 10293801983029384
```

After `rm.all` cache keys still exist and nothing is delete from the filesystem.

### Expected Behavior

Exception: `NotFoundError: No cache entry for test-key found in ./cache`

Works as expected in version 16.1.3

### Environment

- Node: 19.6
- OS: Windows 10

**Comment by peterhirn** [comment]
The problem is [lib/util/glob.js#L6](https://github.com/npm/cacache/tree/main/lib/util/glob.js#L6)

The `pattern` argument is `cache\*(content-*|index-*)` on windows, `cache/*(content-*|index-*)` on linux.

Potential fix

```diff
--- a/lib/util/glob.js
+++ b/lib/util/glob.js
@@ -1,7 +1,9 @@
 'use strict'
 
+const path = require('path')
+
 const { promisify } = require('util')
 const glob = promisify(require('glob'))
 
-const globify = (pattern) => pattern.split('//').join('/')
+const globify = (pattern) => pattern.split(path.sep + path.sep).join(path.sep)
 module.exports = (path, options) => glob(globify(path), options)
```

**Comment by supita** [comment]
I'm having an issue where the `stats` from `cacache.verify` are not properly reported on Windows because the files are not detected.

The root of the problem is the same: `globify` function is not replacing to forward-slashes as `glob` library is expecting to handle Windows paths.

I found that before `globify` was at `lib/verify.js`, ant the replacement [was to correct](https://github.com/npm/cacache/pull/101/files#diff-eeac8636d9b6fc8549cda5929066bc44196ae8b9daac1f414fdd61e2e1340db6R16).

```javascript
const globify = pattern => pattern.split('\\').join('/')
```

But when created the file `lib/util/glob.js` and moved `globiby` there, it [was changed](https://github.com/npm/cacache/pull/141/files#diff-20cceb346ffab6392d2b141017e1aa137df33c1c3ce59d5ccc765171586919f4R6) to 

```javascript
const globify = pattern => pattern.split('//').join('/')
```

which is causing problems when `glob` is used for obtaining stats and cleanup in the case of [`verify`](https://github.com/npm/cacache/blob/479b1352a72ea3a6bc403545e269d3add985c6ee/lib/verify.js#L109)
